java-checks/src/main/java/org/sonar/java/checks/LazyArgEvaluationCheck.java

```diff
@@ -296,7 +297,18 @@ public void visitMethodInvocation(MethodInvocationTree tree) {
 
     private static boolean isGetter(MethodInvocationTree tree) {
       String methodName = tree.methodSymbol().name();
-      return methodName != null && (methodName.startsWith("get") || methodName.startsWith("is"));
+      return methodName.startsWith("get") || methodName.startsWith("is") || isGetterMatchingFieldNameAndType(tree.methodSymbol());
+    }
+
+    private static boolean isGetterMatchingFieldNameAndType(Symbol.MethodSymbol methodSymbol) {
+      Symbol owner = methodSymbol.owner();
+      Type getterReturnType = methodSymbol.returnType().type();
+      return methodSymbol.parameterTypes().isEmpty()
```

Comment:
The `methodSymbol.parameterTypes().isEmpty()` is used to check that the candidate getter has no arguments, anyway such a check is not performed in the existing logic. Should we enforce it, and clean the possible hits on the ITs, or should we leave the existing logic and accept those FNs?
@alban-auzeill